usaspending_api/awards/v2/views/funding.py

```diff
@@ -0,0 +1,140 @@
+                ) as per_toptier_row_number
+            from
+                agency a
+                inner join toptier_agency ta on ta.toptier_agency_id = a.toptier_agency_id
+                left outer join subtier_agency sa on sa.subtier_agency_id = a.subtier_agency_id
+        ) t
+        where t.per_toptier_row_number = 1
+    )
+    select
+        coalesce(gfaba.transaction_obligated_amount, 0.0)              transaction_obligated_amount,
+        taa.agency_id || '-' || taa.main_account_code                  federal_account,
+        fa.account_title,
+        afmap.toptier_agency_name                                      funding_agency_name,
+        afmap.agency_id                                                funding_agency_id,
+        aamap.toptier_agency_name                                      awarding_agency_name,
+        aamap.agency_id                                                awarding_agency_id,
+        oc.object_class                                                object_class,
+        oc.object_class_name                                           object_class_name,
+        pa.program_activity_code                                       program_activity_code,
+        pa.program_activity_name                                       program_activity_name,
+        sa.reporting_fiscal_year,
+        sa.reporting_fiscal_quarter
+    from
+        gather_financial_accounts_by_awards gfaba
+        left outer join treasury_appropriation_account taa on
+            taa.treasury_account_identifier = gfaba.treasury_account_id
+        left outer join federal_account fa on
+            fa.agency_identifier = taa.agency_id and
+            fa.main_account_code = taa.main_account_code
+        left outer join agency_id_to_agency_id_for_toptier_mapping afmap on
+            afmap.toptier_agency_id = taa.funding_toptier_agency_id
+        left outer join agency_id_to_agency_id_for_toptier_mapping aamap on
+            aamap.toptier_agency_id = taa.awarding_toptier_agency_id
+        left outer join object_class oc on
+            gfaba.object_class_id = oc.id
+        left outer join ref_program_activity pa on
+            gfaba.program_activity_id = pa.id
+        left outer join submission_attributes sa on
+            gfaba.submission_id = sa.submission_id
+    group by
+         federal_account, fa.account_title, funding_agency_name, awarding_agency_name, afmap.agency_id, aamap.agency_id,
+         oc.object_class_name, pa.program_activity_name, oc.object_class, pa.program_activity_code,
+         sa.reporting_fiscal_year, sa.reporting_fiscal_quarter, gfaba.transaction_obligated_amount
+    {order_by}
+    limit {limit} offset {offset};
+"""
+)
+
+
+TINYSHIELD_MODELS = customize_pagination_with_sort_columns(list(SORTABLE_COLUMNS.keys()), DEFAULT_SORT_COLUMN)
+TINYSHIELD_MODELS.append(get_internal_or_generated_award_id_model())
+
+
+@validate_post_request(TINYSHIELD_MODELS)
```

Comment:
I'm concerned that without a deepcopy here it is possible for the model state of subsequent requests to be impacted by previous request/responses.

It's a failing of using mutable objects, dictionaries, & lists to define these models. TinyShield models need a refactor to a better definition system. Until then, its safer to use `copy.deepcopy()` when providing the models to Tinyshield.
```python
from copy import deepcopy
```

```suggestion
@validate_post_request(deepcopy(TINYSHIELD_MODELS))
```